test/functional/tests/data_integrity/test_data_integrity_unplug.py

```diff
@@ -0,0 +1,313 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import hashlib
+import os
+import random
+import re
+from bisect import bisect_right
+from datetime import timedelta
+from time import sleep
+
+import pytest
+
+from api.cas import casadm, cli_messages
+from api.cas.cache_config import CacheMode, CleaningPolicy, CacheModeTrait, CacheLineSize
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from storage_devices.ramdisk import RamDisk
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite
+from test_utils.asynchronous import start_async_func
+from test_utils.filesystem.directory import Directory
+from test_utils.output import CmdException
+from test_utils.size import Unit, Size
+
+ram_disk, tmp_dir, fio_seed = None, None, None
+num_jobs = 8
+job_workset_size = Size(1, Unit.MiB)
+block_size = Size(1, Unit.Blocks4096)
+job_workset_blocks = int(job_workset_size / block_size)
+
+total_size = num_jobs * job_workset_size
+total_workset_blocks = int(total_size / block_size)
+
+# g_seqno[b, j] is a list of I/O operations that hit sector b in job j workset.
+# IOs on the list are identified by its sequential number within the job j.
+# g_seqno [b, j] is sorted by ascending I/O number.
+# Value of -1 indicated prefill (meaning no known I/O hit sector b within job
+# j within the analyzed interval)
+g_seqno = [[[-1] for _ in range(job_workset_blocks)] for _ in range(num_jobs)]
+
+# seqno of the last I/O taken into account in g_seqno (same for all jobs)
+max_pattern_io_seqno = -1
+
+
+@pytest.mark.os_dependent
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.asyncio
+async def test_data_integrity_unplug(cache_mode):
+    global fio_seed, tmp_dir, ram_disk
+    cache_dev = TestRun.disks["cache"]
+    core_dev = TestRun.disks["core"]
+
+    sleep_max_s = timedelta(seconds=10)
+
+    with TestRun.step("Test prepare"):
+        random.seed(TestRun.random_seed)
+        fio_seed = random.randint(0, 2 ** 32)
+        TestRun.LOGGER.info(f"FIO seed: {fio_seed}")
```

Comment:
Please generate a unique seed for fio - the same value should be used to all fio calls within single test run